e2etest/pkg/k8s/nodes.go

```diff
@@ -46,15 +46,17 @@ func NodeIPsForFamily(nodes []v1.Node, family ipfamily.Family, vrfName string) (
 			case ipfamily.RequiredDualStack:
 				res = append(res, addr.IPV4Address)
 				res = append(res, addr.IPV6Address)
+			case ipfamily.PreferDualStack:
+				res = append(res, addr.IPV4Address)
+				res = append(res, addr.IPV6Address)
 			}
 			continue
 		}
 		for _, a := range n.Status.Addresses {
 			if a.Type == v1.NodeInternalIP {
-				if family != ipfamily.RequiredDualStack && ipfamily.ForAddress(net.ParseIP(a.Address)) != family {
-					continue
+				if family == ipfamily.RequiredDualStack || family == ipfamily.PreferDualStack || ipfamily.ForAddress(net.ParseIP(a.Address)) == family {
```

Comment:
we should have a wrapping function here, this if is becoming too complex to follow